tools/azure-pipelines/prepare_precommit.sh

```diff
@@ -32,8 +32,17 @@ find . -type f -name '*.timestamp' | xargs touch
 export M2_HOME=/home/vsts/maven_cache/apache-maven-3.2.5/ 
 export PATH=/home/vsts/maven_cache/apache-maven-3.2.5/bin:$PATH
 mvn -version
-mvn install --settings ./tools/azure-pipelines/google-mirror-settings.xml -DskipTests -Drat.skip
+MVN_CALL="mvn install --settings ./tools/azure-pipelines/google-mirror-settings.xml -DskipTests -Drat.skip $PROFILE"
+echo "Invoking Maven: '$MVN_CALL'"
+$MVN_CALL
+EXIT_CODE=$?
 
+if [ $EXIT_CODE != 0 ]; then
+	echo "=============================================================================="
+	echo "Build error. Exit code: $EXIT_CODE. Failing build"
+	echo "=============================================================================="
+	exit $EXIT_CODE
+fi
 
 chmod -R +x build-target
 chmod -R +x flink-end-to-end-tests
```

Comment:
I left this file in, even though it is currently not used. I consider always running the e2e tests an experiment. If we need to undo it, this file is still around.
If the experiment is successful, I will remove it (a cleanup PR is needed anyways once we remove travis support).